### ISSUE TITLE
Colorkey ignored when using `special_flags` in `blit`

### ISSUE THREAD

**Issue by Matiiss** [pygame/pygame#3233]
MRE:
```py
import pygame


WIDTH, HEIGHT = 640, 360
FPS = 60

screen = pygame.display.set_mode((WIDTH, HEIGHT))
clock = pygame.time.Clock()

surf = pygame.Surface((64, 32))
surf.fill("white")
surf.set_colorkey("white")

running = True
while running:
    clock.tick(FPS)
    screen.fill("black")

    events = pygame.event.get()
    for event in events:
        if event.type == pygame.QUIT:
            running = False

    screen.blit(surf, (320 - 64, 180 - 16))
    screen.blit(surf, (320, 180 - 16), special_flags=pygame.BLEND_MAX)

    pygame.display.flip()
```
Result:
![image](https://user-images.githubusercontent.com/83066658/173750344-336ab3e3-69dd-426b-94b4-04c86d5a9b11.png)

The expected result would be an all-black surface because the colorkey should be respected regardless of flags, as I imagine at least. It seems to be ignored with flags, tested for similar behavior with `pygame.BLEND_MAX`, a lighter background fill and darker surface fill.

**Comment by MyreMylar** [comment]
I recommend just using 0 alpha pixels and an RGB blend. The special blend
modes don't support colorkeys and never have afaik.

Also, colorkeys are slower these days because of vectorisation of
operations. If you have to test if each pixel is colour X before blitting
it then you can't do the same operation on 16 colour channels
simultaneously as cheaply.

TLDR use alpha pixels instead of colorkeys.

**Comment by Matiiss** [comment]
@MyreMylar So, if I were to use `pygame.BLEND_RGB_MIN` when blitting an opaque surface on a surface created with `pygame.SRCALPHA` flag that was filled with `(0, 0, 0, 0)` and then had a `(255, 255, 255, 0)` polygon drawn on it, then it shouldn't care about those alpha values and only look at the RGB, right? So essentially where it would be drawn on with that white transparent color there that blitted surface would be completely visible, right?

**Comment by MyreMylar** [comment]
Sounds right, but I'd run a test program to test it behaves as you expect.

**Comment by Matiiss** [comment]
Ok, so I guess this won't be worked on then, right? It's gonna be left as is (although at least as a proposal I could suggest such an enhancement for future, that blend modes support colorkeys, but if not then fine).

**Comment by MyreMylar** [comment]
If someone wants to add support in the future it's possible, but honestly with ever increasing vectorization at the colour channel level I think a color key is always going to be slower than using alpha for the same effect so there isn't that much incentive to add a slower path to do something you can already do.

Color key mainly exists for backwards compatibility with the pygame 1 era when it was an optimization to test each pixel to see if it matched a specific colour and skip drawing entirely if it did. You could sidestep expensive alpha blending if you were happy with cut outs. 

Now that optimization is probably a wash for the 'cut out' case because the colour comparison itself is expensive and has to be done per pixel, meanwhile blending operations have gotten faster because they can be done simultaneously per channel (i.e across multiple pixels). I guess I'd have to see tests to compare these days.

In the world of special blendmodes you are already operating a blend on every non-cut out pixel anyway (otherwise why are you doing the blend?) so the cut out pixel comparison is now on top of the blend operation for every non-cut out pixel which is pretty much certain to slow it down versus just doing the blend for every pixel. I expect that for several of the blend modes (e.g. add) the blend itself is faster than a pixel comparison - even without vectorization.

Anyway, TLDR - I'm not sure how much worth the colorkey cut out technique has nowadays versus just using an alpha channel which also gives you nice, optional, semi-alpha feathered edges on your cut outs.

**Comment by Matiiss** [comment]
Ah, ok, thanks for the insight, ok, then I'll close the issue and thanks for the help!
